packages/react/src/Hero/Hero.tsx

```diff
@@ -1,69 +1,90 @@
-import React, {forwardRef} from 'react'
+import React, {forwardRef, PropsWithChildren} from 'react'
 import clsx from 'clsx'
 import styles from './Hero.module.css'
 import {Button} from '../Button'
-import {Heading} from '../Heading'
+import {Heading, HeadingProps, HeadingTags} from '../Heading'
 import {Text} from '../Text'
 
 import type {BaseProps} from '../component-helpers'
 import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/hero/base.css'
 
-type Action = {
-  text: string
+export type HeroProps = BaseProps<HTMLDivElement> & {
+  align?: 'start' | 'center'
+  children: React.ReactNode | React.ReactNode[] // Maybe use this from `FAQ`? type FAQRootProps = PropsWithChildren<BaseProps<HTMLElement>>
+}
+
+const Root = forwardRef<HTMLDivElement, HeroProps>(({className, align = 'start', children, ...rest}, ref) => {
+  return (
+    <section
+      className={clsx(styles.Hero, styles[`Hero--align-${align}`], className)}
+      ref={ref}
+      aria-labelledby="hero-section-brand-heading"
+      {...rest}
+    >
+      {React.Children.map(children, child => {
+        if (React.isValidElement(child)) {
+          if (child.type === HeroHeading || child.type === HeroDescription) {
+            return child
+          } else if (child.type === HeroPrimaryAction || child.type === HeroSecondaryAction) {
+            return <div className={styles['Hero-actions']}>{child}</div>
+          } else {
+            return child
+          }
+        }
+      })}
+    </section>
+  )
+})
+
+type HeroHeadingProps = {
+  as?: HeadingTags['as']
+  children: React.ReactNode | React.ReactNode[]
+} & HeadingProps
+
+const HeroHeading = forwardRef<HTMLHeadingElement, HeroHeadingProps>(({as = 'h1', children, ...rest}, ref) => {
+  return (
+    <Heading id="hero-section-brand-heading" className={styles['Hero-heading']} as={as} ref={ref} {...rest}>
+      {children}
+    </Heading>
+  )
+})
+
+type HeroDescriptionProps = {
+  children: React.ReactNode | React.ReactNode[]
+}
+
+function HeroDescription({children}: PropsWithChildren<HeroDescriptionProps>) {
```

Comment:
Should we allow any props to be passed down to the `<Text>` component like we do with `HeroHeading`, or should we keep this locked down?

Reply:
For descriptions, we keep them locked down atm (see Card, SectionIntro or the upcoming [Pillar](https://github.com/primer/brand/pull/262/files#diff-338f2497ae157e09ae2127ab08a052c0bae3161ab0cda467a64451efb5a1945fR90) examples).

Reply:
I think for Hero we may need to be a bit more flexible @josepmartins. The previous Hero implementation was too restrictive, because you couldn't granularly control aspects of the text to meet requirements around brand expression. Hero's are unique in this sense. 

I think we should forward the `weight` and `size` props as an escape hatch. These are really useful, but we don't need to encourage it in documentation. It's there only if needed.

Reply:
Thanks for the feedback y'all! I implemented both `weight` and `size`, while keeping the other props locked down. I added a small note in the documentation for it under the props table only explaining that it's available.